src/sentry/eventstore/utils.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
+
+import logging
+
+methods_to_test = [
+    "get_next_event_id",
+    "get_prev_event_id",
+    "get_earliest_event_id",
+    "get_latest_event_id",
+]
+
+logger = logging.getLogger("sentry.eventstore")
+
+
+def selector_func(context, method, callargs):
+    if method in methods_to_test:
+        return ["snuba", "snuba_discover"]
+
+    return ["snuba"]
+
+
+def callback_func(context, method, callargs, backends, results):
+    """
+    Log if results are different
+    """
+    if backends == ["snuba", "snuba_discover"] and method in methods_to_test:
+        if results[0].result() != results[1].result():
```

Comment:
Is this logic temporary? Because there are many cases where the results will not be exactly the same since the table is being updated with new events while you run the two queries. So a mismatch coefficient like we did for outcomes seem a better idea to identify when we are doing good.

Reply:
It's temporary in that we won't be running both queries, logging differences or using Service Delegator once we feel that Discover is stable. What are you proposing with mismatch coefficient exactly?

I feel like we can tweak this logging if it's too noisy / not giving us useful data later, but would rather start with something basic that just captures all of the data since:
- Not sure if timing is a real issue since queries are rounded down to the second anyway
- We expect differences in the 2 implementations and many of these differences are desired behavior
- It seems very possible that this might only happen very infrequently and we can just inspect each of the differences

Reply:
ok